Dashboard/app/js/lib/controllers/permissions.js

```diff
@@ -259,6 +259,30 @@ FLOW.permControl = Ember.Controller.create({
       }
       return false;
   }.property(),
+
+  userCanViewData: function (formID) {
+    var permissions = this.formPermissions(formID);
+    return permissions.indexOf("DATA_READ") >= 0;
+  },
+
+  formPermissions: function (formId) {
```

Comment:
if we pass in the `form` object itself instead of the `formId` only, we could skip the extra loop in the `FLOW.surveyControl` to get the `form` object.

Reply:
It also means you would be able to reuse the `FLOW.permControl.permissions(form)` function to get its permissions instead of rewriting that code.